app/about/side-nav.tsx

```
@@ -0,0 +1,85 @@
+'use client';
+import React, { useEffect, useState } from 'react';
+import { SideNav } from '@trussworks/react-uswds';
+
+function useActiveSection(sectionIds: string[]) {
+  const [activeSection, setActiveSection] = useState<string | null>(null);
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(
+      (entries) => {
+        const visibleSections = new Map<string, number>();
+
+        entries.forEach((entry) => {
+          if (entry.isIntersecting) {
+            const distance = Math.abs(entry.boundingClientRect.top);
+            visibleSections.set(entry.target.id, distance);
+          }
+        });
+
+        if (visibleSections.size > 0) {
+          const closestSection = Array.from(visibleSections.entries()).reduce(
+            (closest, current) => {
+              return current[1] < closest[1] ? current : closest;
+            },
+          );
+          setActiveSection(closestSection[0]);
+        }
+      },
+      {
+        rootMargin: '-10% 0px -70% 0px',
+        threshold: [0, 0.25, 0.5, 0.75, 1],
+      },
+    );
+
+    sectionIds.forEach((id) => {
+      const element = document.getElementById(id);
+      if (element) observer.observe(element);
```

Should we add an `else` to warn about the missing element? For example 
```
console.warn(`Element with ID '${id}' not found in the DOM);
```
Might be better then just silently skipping, for debugging.

---

Ups, you apparently did not see this comment. Maybe not that important though, now that we generate the list of ids from actual page content.